app/Dockerfile.base

```diff
@@ -16,6 +11,16 @@ RUN apt-get update && apt-get install -y \
     libspatialindex-dev \
     libgdal1-dev
 
+# Dependencies for installing docker locally
+RUN apt-get update && apt-get install -y \
+    curl
+RUN curl -L https://get.docker.com/builds/Linux/x86_64/docker-1.10.3 > /usr/bin/docker && chmod +x /usr/bin/docker
```

Comment:
It may be useful to pull the `1.10.3` bit out into an environment variable so that it is easier to update later.